test/acceptance/expected_invocation_count_test.rb

```diff
@@ -226,4 +226,15 @@ def test_should_fail_fast_if_there_is_no_matching_expectation
       '- expected exactly once, invoked never: #<Mock:mock>.method(1)'
     ], test_result.failure_message_lines
   end
+
+  def test_should_pass_if_cardinality_is_satisfied_using_calls_made_both_before_and_after_updating_cardinality
+    test_result = run_as_test do
+      mock = mock('mock')
+      expectation = mock.expects(:method)
+      mock.method
+      expectation.twice
+      mock.method
+    end
+    assert_passed(test_result)
+  end
```

Comment:
Excellent - thanks for adding an acceptance test. I think it does make sense to make this behaviour part of the public API.

Reply:
> To me actually, it only _kinda_ makes sense. Updating expectations after the SUT has already been (partly - or even worse, fully) exercised seems like an odd thing to do that should, in theory, be disallowed or discouraged. I'm not sure if that's possible or desirable in practice, though.
> 
> So, pragramatically, this seems like the _most_ sensible even if not the _fully_ sensible choice.

I see what you mean. However, I don't feel strongly enough that it should be disallowed or even discouraged, so I'm happy to stick with what you've done in this PR.